### DISCUSSION TITLE
Define detector at camera level

### DISCUSSION THREAD

**Discussion by clsferguson** [blakeblackshear/frigate#15604] [Ideas]
It's probably an edge case, but I was just thinking it would be nice to define the detector at the camera level, allowing the use of multiple detectors. It may become more prevalent if the coral is no longer made, and people find themselves needing another detector, but don't want to retire their already working coral. Users could set their 1080P cams to the coral and their 4k cams on a tensorrt kind of thing.

**Comment by NickM-27** [comment]
If the models are different this gets really difficult. And it would change the paradigm for how detectors work which right now is that one detector can serve all cameras.

**Comment by santiagordguez** [comment]
I agree with @clsferguson. It would be nice to execute detectors only in selected cameras even for saving resources.

**Reply by NickM-27** [reply]
You can disable detect for cameras that you don't want to run detection on

**Reply by NickM-27** [reply]
I just saw your other post. 

> I don't want to consume resources looking for objects that will never appear in some of the cameras.

Your understanding is incorrect. For each region Frigate sends one object detection inference to detect all objects that the model is capable of detecting. There are not separate requests for each object type.

**Comment by plysdyret** [comment]
It would be cool though, different detectors/models might fit different camera setups better, and different model resolutions also
